### DISCUSSION TITLE
May I custom MarkerType for Edge?

### DISCUSSION THREAD

**Discussion by zilong** [xyflow/xyflow#2375] [New Features]
not only MarkerType.Arrow ,  MarkerType.ArrowClosed， May I custom more markerType for edge?

**Comment by bcakmakoglu** [comment]
Yes, you can pass an object of type `EdgeMarker` to `markerEnd` or `markerStart`.
[Here's the relevant docs.](https://reactflow.dev/docs/api/edges/edge-options/#markerstart--markerend-options)

**Reply by zilong** [reply]
markerEnd: {
      type: MarkerType.ArrowClosed,  ==> can't make more type!
      color: '#FFCC00',  
      orient: 'auto-start-reverse',
      markerUnits: 'userSpaceOnUse',
      width: 20,
      height: 20,
    },

**Reply by zilong** [reply]
export const MarkerSymbols = {
  [MarkerType.Arrow]: ArrowSymbol,
  [MarkerType.ArrowClosed]: ArrowClosedSymbol,
};

export function useMarkerSymbol(type: MarkerType) {
  const symbol = useMemo(() => {
    const symbolExists = MarkerSymbols.hasOwnProperty(type);

    if (!symbolExists) {
      // @ts-ignore
      if (process.env.NODE_ENV === 'development') {
        console.warn(`[React Flow]: Marker type "${type}" doesn't exist. Help: https://reactflow.dev/error#900`);
      }
      return () => null;
    }

    return MarkerSymbols[type];
  }, [type]);
  return symbol;
}

**Comment by moklick** [comment]
If you want to use a custom marker you need to add an SVG with a marker object and use its id as a string for `markerEnd` or `markerStart`.The docs say "string (id for a svg marker that you need to define yourself) or a [marker configuration object](https://reactflow.dev/docs/api/edges/edge-options/#markerstart--markerend-options)". Should we explain it better? Maybe an example would be a good idea here..

**Reply by zilong** [reply]
got It! Maybe  React-Flow could provide a more friendly API?

**Reply by martypitt** [reply]
Hi.  Thanks for the info.  How do we pass the SVG into react flow?  The actual svg element is abstracted away, so not sure how to inject the element into the def tag?

**Reply by martypitt** [reply]
Just found [this comment](https://github.com/wbkd/react-flow/discussions/1658#discussioncomment-2729583), which answers my question.

Thanks again.
